src/devices/src/virtio/rng/persist.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines the structures needed for saving/restoring entropy devices.
+
+use snapshot::Persist;
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use vm_memory::GuestMemoryMmap;
+
+use crate::virtio::persist::Error as VirtioStateError;
+use crate::virtio::rng::{Entropy, Error as EntropyError, NUM_QUEUES, QUEUE_SIZE};
+use crate::virtio::{VirtioDeviceState, TYPE_RNG};
+
+#[derive(Clone, Versionize)]
+pub struct EntropyState {
+    virtio_state: VirtioDeviceState,
+}
```

Comment:
```suggestion
#[derive(Clone, Versionize)]
pub struct EntropyState(VirtioDeviceState);
```

Reply:
We can only derive `Versionize` on structs with named fields.